homework01/rudolf/vector.cpp

```diff
@@ -0,0 +1,46 @@
+#include "vector.h"
+
+Vector::Vector()
+{
+	mX=0;
+	mY=0;
+}
+
+Vector::Vector(double x, double y): mX(x), mY(y){}
+
+double Vector::dotProduct(const Vector& vec) const
```

Comment:
Also: in general we like writing as little code as possible - the best code is the one that does not exist.